lms/djangoapps/grades/tests/test_tasks.py

```diff
@@ -149,11 +149,9 @@ def test_query_counts(self, default_store, num_mongo_calls, num_sql_calls):
                 with self.assertNumQueries(num_sql_calls):
                     self._apply_recalculate_subsection_grade()
 
-    # TODO (TNL-6225) Fix the number of SQL queries so they
-    # don't grow linearly with the number of sequentials.
     @ddt.data(
-        (ModuleStoreEnum.Type.mongo, 1, 46),
-        (ModuleStoreEnum.Type.split, 3, 45),
+        (ModuleStoreEnum.Type.mongo, 1, 24),
+        (ModuleStoreEnum.Type.split, 3, 23),
```

Comment:
I'm not sure why these aren't 21 and 20 like the tests above, will dig a bit further on that.